### PR TITLE
Linux AppImage + GH Actions

### PR DESCRIPTION
**Pull request by rithvikvibhu**
This PR adds AppImage packaging for Linux.

### README
Both the main and `builds/linux/` readme updated.

### Structure
An FHS-like path is needed, placed in `builds/linux/appdir`.
`fingertip` and `hnsd` binaries must be placed in `builds/linux/appdir/usr/bin`. Other files like `.desktop` and icons are in their respective locations.

### Desktop integration
Icon loads properly:
image

Taskbar:
image

### Icons
Desktop files require png only, so extracted from the existing `fingertip.ico` file.

### GH Action
Triggers on push and for pull requests. Builds on Ubuntu 18.04, so usable pretty much everywhere. Tested on `Manjaro 21.1.0`.

**Note:** Builds depend on the glibc in your system. This build will only work on systems with glibc >= current version. This is why the CI builds in an old Ubuntu 18.04 container. [More info](https://github.com/probonopd/linuxdeployqt#a-note-on-binary-compatibility)

`linuxdeployqt` may throw an error if glibc is too new (yes, new). Add `-unsupported-allow-new-glibc` to bypass it.

### CODE REVIEW

**Comment by rithvikvibhu** [comment]
I guess the workflow will only work after it is merged. Check out the same run on my fork: https://github.com/rithvikvibhu/fingertip/actions/runs/1089789435

`fingertip-appimage` is the build result. (github downloads it compressed in a zip file even if it's a single file ¯\_(ツ)_/¯

File: .github/workflows/build-linux.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Checkout hnsd repository
+        uses: actions/checkout@v2
+        with:
+          repository: 'handshake-org/hnsd'
+
+      - name: Install dependencies
+        run: sudo apt install -y unbound libunbound-dev
```
Comment (buffrr) [inline .github/workflows/build-linux.yml:16]: Is there a reason for installing both `unbound` and `libunbound-dev` here?

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,10 @@
 .idea
-rsrc.syso
\ No newline at end of file
+rsrc.syso
+
+# Linux build and package
+builds/linux/appdir/*
```
Comment (buffrr) [inline .gitignore:5]: Nit: could you move these to a `.gitignore` file in `builds/linux` directory similar to `builds/windows`? 
Also the .gitkeep in the `appdir/usr/bin/.gitkeep` directory can be replaced with `.gitignore` with the rules `*`, `!.gitignore`

**Review by buffrr — CHANGES_REQUESTED** [review]

File: .github/workflows/build-linux.yml (hunk 1 above)
Comment (rithvikvibhu) [reply .github/workflows/build-linux.yml]: Removed unbound.

File: .gitignore (hunk 2 above)
Comment (rithvikvibhu) [reply .gitignore]: Wow TIL that `.gitkeep` isn't really a git thing. Replaced with `.gitignore` and moved the other one to `build/linux/`

**Comment by buffrr** [comment]
Thanks for the PR! Everything looks good so far can you rebase it?

**Comment by rithvikvibhu** [comment]
Hey sorry, missed this email. Came here to check and found your comment.
It was already based on the latest commit, so squashed it and tested the latest build again (https://github.com/rithvikvibhu/fingertip/actions/runs/1151339387) and looks like it works normally.
(no, I didn't generate that vanity commit hash :p)

**Review by buffrr — APPROVED** [review]

**Pull request merged**